### PR TITLE
Parse the AY field  in the openAIR file

### LINKED ISSUE #1118: OpenAIR Parser: Parse the AY field
What should XCSoar do differently, what functionality should be added?
----------------------------------------------------------------------
Parse the AY field and display it in the Airspace popup when selecting an airspace

```
AY = Airspace Type (extends AC which should now only be used for class)
```

Example Airspace:
```
AC E
AY TMA
AN TMA BALE T1 HX
AI 62e10d61d0d055fccf50f375
AL 5200ft MSL
AH 5700ft MSL
DP 47:25:52 N 007:46:41 E
DP 47:19:12 N 007:51:31 E
DP 47:17:28 N 007:48:42 E
DP 47:19:05 N 007:26:21 E
DP 47:24:44 N 007:22:28 E
DP 47:22:60 N 007:39:30 E
DP 47:25:52 N 007:46:41 E
```

### PR DESCRIPTION
**Pull request by ubx**
Brief summary of the changes
----------------------------
This PR implements parsing the AY field in the openAIR file and display it in the Airspace popup

Related issues and discussions
------------------------------

Closes  #1118

### CODE REVIEW

**Comment by lordfolken** [comment]
Can you edit you squash your fixup commit and do a force push? (git rebase -i HEAD~5)

File: src/Airspace/AirspaceParser.cpp [hunk 1]
```diff
@@ -55,13 +55,13 @@ enum class AirspaceFileType {
 struct AirspaceClassCharCouple
 {
   const TCHAR character;
-  AirspaceClass type;
+  AirspaceClass asclass;
```
Comment (MaxKellermann) [inline src/Airspace/AirspaceParser.cpp:58]: Why rename those fields? This makes the diff more confusing than it needs to be

File: test/src/harness_airspace.cpp [hunk 2]
```diff
@@ -39,12 +39,12 @@
 static void
 airspace_random_properties(AbstractAirspace& as)
 {
-  AirspaceClass Type = (AirspaceClass)(rand()%14);
+  AirspaceClass asclass = (AirspaceClass)(rand()%14);
```
Comment (MaxKellermann) [inline test/src/harness_airspace.cpp:42]: Why rename?

**Comment by lordfolken** [comment]
Please be aware of the header: 
```
* This is OpenAIR file is an extended format derived from the original specification. It adds additional fields:
*
*  - AY: The AY field contains the type and the AC field is now exclusively used to specify the ICAO class of the defined airspace.
*  - AF: The AF field contains contact frequencies defined for the airspace. Frequencies are exported as "name frequency MHz" delimited by ";". If no frequencies are defined, "NIL" is used instead.
*
* This OpenAIR file does contain non-standard AY values. The following list shows all AY types that may be used within this file:
* UNCLASSIFIED, RESTRICTED, DANGER, PROHIBITED, CTR, TMZ, RMZ, TMA, TRA, TSA, FIR, UIR, ADIZ, ATZ, MATZ, AWY, MTR, ALERT, WARNING, PROTECTED, HTZ, GLIDING_SECTOR, TRP, TIZ, TIA, MTA, CTA, ACC_SECTOR, AERIAL_SPORTING_RECREATIONAL, OVERFLIGHT_RESTRICTION
*
* This OpenAIR file does contain non-standard AC values. The following list shows all AC types that may be used within this file:
* A, B, C, D, E, F, G, UNCLASSIFIED
```

I guess we still need a few class types.

**Comment by lordfolken** [comment]
Why did you close?

**Comment by ubx** [comment]
Because renaming seems to be a problem

**Comment by MaxKellermann** [comment]
What the ......?
Yet more of my time wasted on reviewing a PR that will never be finished.

**Comment by lordfolken** [comment]
if you have to rename and its a necessary change ((which i think it is) due to the changed use of terminology used in the standard) then this should be an individual commit, and the reason should be explained in the commit message.

If you put everything in one commit it becomes huge and the reasons for it is not transparent. 

Each transformation to the code should be atomic and explain the reasoning behind.

**Comment by lordfolken** [comment]
I'm sorry you actually did mention it:

The AY is parsed as Airspace Type and the AC is parsed as Airspace Class.
Add new field astype for airspace type, is not interpreted, just displayed on the Airspace Details popup.
To avoid naming confusion, renamed 'type' to 'asclass', 'getType' to 'getClass' and 'TempAirspaceType' to 'TempAirspace'

**Comment by ubx** [comment]
To not waste my time, is it ok if I would it split into two PRs ?: 
1. solely renaming necessary variables and methods as mentioned in the commit message
2. implement the functionality to parse the AY field in the openAIR file

**Pull request closed without merging**